Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/cmd/kubelet/app/server.go

```diff
@@ -241,52 +241,45 @@ func (s *KubeletServer) AddFlags(fs *pflag.FlagSet) {
 	fs.BoolVar(&s.Containerized, "containerized", s.Containerized, "Experimental support for running kubelet in a container.  Intended for testing. [default=false]")
 }
 
-// Run runs the specified KubeletServer.  This should never exit.
-func (s *KubeletServer) Run(_ []string) error {
-	util.ReallyCrash = s.ReallyCrashForTesting
-	rand.Seed(time.Now().UTC().UnixNano())
-
-	// TODO(vmarmol): Do this through container config.
-	if err := util.ApplyOomScoreAdj(0, s.OOMScoreAdj); err != nil {
-		glog.Warning(err)
-	}
-
-	client, err := s.createAPIServerClient()
-	if err != nil && len(s.APIServerList) > 0 {
-		glog.Warningf("No API client: %v", err)
+// KubeletConfig generates the appropriate Kubelet config for the given server arguments, or returns an
+// error.
+func (s *KubeletServer) KubeletConfig() (*KubeletConfig, error) {
```

Comment:
Just based on names, doesn't this seem backwards?  A `KubeletServer` produces a `KubeletConfig`?